lib/preexec.bash

```diff
@@ -23,3 +23,65 @@ set +T
 
 # Modify `$PROMPT_COMMAND` now
 __bp_install_after_session_init
+
+## Helper functions
+function __check_precmd_conflict() {
+	local f # TODO: trim whitespace like preexec does
+	for f in "${precmd_functions[@]}"; do
+		if [[ "${f}" == "${1}" ]]; then
+			return 0
+		fi
+	done
+	return 1
+}
+
+function __check_preexec_conflict() {
+	local f # TODO: trim whitespace like preexec does
+	for f in "${preexec_functions[@]}"; do
+		if [[ "${f}" == "${1}" ]]; then
```

Comment:
this is code dup, maybe use `_bash-it-array-contains-element`?

Reply:
Yeah, it should use the trim white spec function and then the array function for clarity and consistency. I just didn't get that far in to it. 

I'll update it when I'm back at my computer!

Reply:
Done and done.